Orange/widgets/visualize/owpythagorastree.py

```diff
@@ -0,0 +1,416 @@
+import math
+from collections import namedtuple, defaultdict
+
+import Orange
+from Orange.widgets import gui
+from Orange.widgets.widget import OWWidget
+from Orange.widgets.settings import Setting
+from Orange.classification.tree import TreeClassifier
+from Orange.data.table import Table
+
+from PyQt4 import QtGui, QtCore
+from PyQt4.QtCore import Qt
+
+PI = math.pi
+
+# Please note that all angles are in radians
+Square = namedtuple('Square', ['center', 'length', 'angle'])
+Point = namedtuple('Point', ['x', 'y'])
+
+
+class OWPythagorasTree(OWWidget):
+    name = "Pythagoras Tree"
+    description = "Generalized Pythagoras Tree for visualizing trees."
+    priority = 100
+
+    inputs = [("Classification Tree", TreeClassifier, "set_ctree")]
+    outputs = [("Selected Data", Table)]
+
+    # Settings
+    zoom = Setting(5)
+
+    def __init__(self):
+        super().__init__()
+
+        # Instance variables
+        self.raw_tree = None
+        self.tree = None
+
+        # GUI - CONTROL AREA
+        layout = QtGui.QFormLayout()
+        layout.setVerticalSpacing(20)
+        # layout.setFieldGrowthPolicy(layout.ExpandingFieldsGrow)
+
+        box = self.display_box = \
+            gui.widgetBox(self.controlArea, "Display", addSpace=True,
+                          orientation=layout)
+        layout.addRow(
+            "Zoom ",
+            gui.hSlider(box, self, 'zoom',
+                        minValue=1, maxValue=10, step=1, ticks=False,
+                        callback=None,
+                        createLabel=False, addToLayout=False, addSpace=False))
+        # Stretch to fit the rest of the unsused area
+        gui.rubber(self.controlArea)
+
+        # GUI - MAIN AREA
+        self.scene = QtGui.QGraphicsScene(self)
+        self.view = QtGui.QGraphicsView(self.scene, self.mainArea)
+
+        # Flip y axis for sane drawing
+        matrix = QtGui.QMatrix()
+        matrix.scale(1, -1)
+        self.view.setMatrix(matrix)
+
+        # self.view.setAlignment(Qt.AlignBottom | Qt.AlignHCenter)
+        self.view.setVerticalScrollBarPolicy(Qt.ScrollBarAlwaysOff)
+        self.view.setHorizontalScrollBarPolicy(Qt.ScrollBarAlwaysOff)
+        self.view.setRenderHint(QtGui.QPainter.Antialiasing)
```

Comment:
If these are static properties of QGraphicsView, you should set them as kwargs to the constructor. See: https://github.com/biolab/orange3/blob/master/CONTRIBUTING.md#coding-style